ufpy/algebra/sets.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+from os import environ
+
+from ufpy.math_op import i_generator, r_generator
+from ufpy.utils import withrepr
+
+__all__ = (
+    'USet',
+    'U'
+)
+
+from typing import Iterable, Iterator, Optional, TypeVar, overload
+
+T = TypeVar('T')
+OT = TypeVar('OT')
+
+@i_generator
+@r_generator
+class USet[T]:
+    @overload
+    def __init__(self, *values: T) -> None: ...
+    @overload
+    def __init__(self, *values: T, auto_update_U: bool) -> None: ...
+    @overload
+    def __init__(self, *, iterable: Iterable[T]) -> None: ...
+    @overload
+    def __init__(self, *, iterable: Iterable[T], auto_update_U: bool) -> None: ...
+
+    def __init__(self, *values: T, iterable: Optional[Iterable[T]] = None, auto_update_U: bool = True) -> None:
+        self.__set = set(iterable) if iterable else set(values)
+        self.__auto_update_U = auto_update_U
+        self.__update__()
+        
+    @property
+    def set(self) -> set[T]:
+        return self.__set
+    
+    @set.setter
+    def set(self, value: Iterable[T]):
+        self.__set = set(value)
+        self.__update__()
+        
+    # When USet updates (not required, that new USet must be not same that old one)
+    def __update__(self):
+        if self.__auto_update_U:
+            U(iterable=_get_U() | self)
+    
+    # Convert to other types
+    def __repr__(self) -> str:
+        return f'u{self.set if self.set else '{}'}'
+    
+    def __str__(self) -> str:
+        return repr(self.set) if self.set else '{}'
+    
+    # Iteration
+    def __iter__(self) -> Iterator[T]:
+        return iter(self.__set)
+    
+    # Or
+    def __or__(self, other: Iterable[OT]) -> USet[T | OT]:
+        return USet(iterable=set(self) | set(other), auto_update_U=self.__auto_update_U)
+
+    def __add__(self, other: Iterable[OT]) -> USet[T | OT]:
+        return self | other
+
+    # Substract
+    def __sub__(self, other: Iterable[OT]) -> USet[T]:
+        return USet(iterable=set(self) - set(other), auto_update_U=self.__auto_update_U)
+    
+    def __div__(self, other: Iterable[OT]) -> USet[T]:
```

Comment:
No, in algebra we use \ operator for subtraction, but in Python there isn't ability to create support for \ operator (because of this operator doesn't exists). I took similar operator.